### PR TITLE
Fixed the inconsistence of column name on the list of roles page

### PR DESCRIPTION
**Pull request by RyanL1997**
Signed-off-by: Ryan Liang <jiallian@amazon.com>

### Description
Fixed the column name "Index permissions" on the list of roles page is inconsistent with the field name "Index" on the "Create Role" page

### Issues Resolved
* Resolves https://github.com/opensearch-project/security-dashboards-plugin/issues/922

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peternied — CHANGES_REQUESTED** [review]
* Your commit author and sign off don't look like they are using your legit email address associated with GitHub, could you please update it, see [this for help](http://treeindev.net/article/git-change-commit-name#:~:text=4%20Easy%20Steps%20to%20Change%20Author%20Name%20of,commit%20--amend%20--author%3D%22Author%20%22.%20...%20More%20items...%20).

* Please update the description on this pull request to remove the instructions and condense it as much as possible (I don't include the category because its not very useful for reviewers as an example)

* When you associate issues, use `* Resolves [ISSUE#]` this will automatically close the issue when your pull request is merged.


See a full example of this in https://github.com/opensearch-project/security-dashboards-plugin/pull/940 

* Please investigate the CI failure and report the findings on this pull request if it needs to be retried or delayed while another PR is merged.

**Comment by cwperks** [comment]
@RyanL1997 A change was merged into main to resolve the CI failures. Can you please rebase?

**Comment by RyanL1997** [comment]
@cwperks Thx for the information. I will rebase it.

**Comment by RyanL1997** [comment]
Hi @peternied, Thank you so much for the review with all these clear instructions! I have updated my signoff and recommitted it. The CI failing got fixed for now.

**Review by peternied — APPROVED** [review]
Perfect thanks!

File: public/apps/configuration/panels/role-list.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ const columns: Array<EuiBasicTableColumn<RoleListing>> = [
   },
   {
     field: 'indexPermissions',
```
Comment (cwperks) [inline public/apps/configuration/panels/role-list.tsx:73]: We may want to consider a change to the variable name as well to be consistent.

**Review by cwperks — APPROVED** [review]

**Comment by devardee** [comment]
@RyanL1997  @cliu123 shouldn't it be `Index Pattern` instead ?

**Comment by cliu123** [comment]
> * [BUG] The column name "Index permissions" on the list of roles page is inconsistent with the field name "Index" on the "Create Role" page #922

The value of the field can be index name or index pattern. Whatever it is, it always represents index/indices. So "Index" should be fine. There was a green light from PM. The key is to make the column name and the field name consistent.

**Pull request merged**